Sources/Xcodeproj/pbxproj().swift

```diff
@@ -562,12 +552,12 @@ func xcodeProject(
                     var moduleMapGenerator = ModuleMapGenerator(for: clangTarget, fileSystem: fileSystem)
                     try moduleMapGenerator.generateModuleMap(inDir: path)
                     moduleMapPath = path.appending(component: moduleMapFilename)
-                    isGenerated = true
 
                     includeGroup.addFileReference(path: moduleMapPath.asString, name: moduleMapPath.basename)
-                    // Save this modulemap path mapped to module so we can later wire it up for its dependees.
-                    modulesToModuleMap[target] = (moduleMapPath, isGenerated)
+                    targetSettings.common.MODULEMAP_FILE = "$(SRCROOT)/" + moduleMapPath.relative(to: sourceRootDir).asString
                 }
+                targetSettings.common.CLANG_ENABLE_MODULES = "YES"
+                targetSettings.common.DEFINES_MODULE = "YES"
```

Comment:
It does not set `DEFINES_MODULE=YES` if user defined modulemap exists, since I don't know about the issue mentioned in following comment:
```swift
            // Disable defines target for clang target because our clang targets are not proper framework targets.
            // Also see: <rdar://problem/29825757> 
            targetSettings.common.DEFINES_MODULE = "NO"
```